### PR TITLE
[FAB-17553] Rolling Upgrade scenario from 1.4 to 2.0

### PR DESCRIPTION
**Pull request by suryalnvs**
This PR enables running upgrade scenario as part of weekly run
on every friday. regression/weekly/upgrade2.0.sh script has
the steps for upgrade process.

It can be executed by running: make regression/upgrade

Signed-off-by: Surya Lanka <suryalnvs@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,6 +20,9 @@ include gotools.mk
 regression/%: pre-reqs
 	cd ${@} && ginkgo -v
 
+regression/upgrade:
+	cd weekly && ./upgrade2.0.sh 1.4-stable 2.0-stable
```
Comment (lindluni) [inline Makefile]: Needs to be `cd regression/weekly`

**Comment by lindluni** [comment]
Can we change the name of the `weekly` directory to `upgrade`

File: tools/operator/scripts/upgradeNetwork.sh [hunk 2]
```diff
@@ -106,11 +106,19 @@ modifyAndSubmit(){
...
-  export CORE_PEER_FILESYSTEMPATH=$PWD
-  peer node upgrade-dbs
-  cd -
+  docker run --name peer-cli --rm \
+  -e CORE_PEER_LOCALMSPID=$1 \
+  -e CORE_PEER_TLS_ENABLED=true \
+  -e CORE_PEER_MSPCONFIGPATH=/etc/hyperledger/fabric/artifacts/users/Admin@$3/msp \
+  -v $4/backup/$2/:/var/hyperledger/production/ \
+  -v $4/crypto-config/peerOrganizations/$3/:/etc/hyperledger/fabric/artifacts/ \
+  hyperledger-fabric.jfrog.io/fabric-peer:amd64-2.0-stable peer node upgrade-dbs
```
Comment (lindluni) [inline tools/operator/scripts/upgradeNetwork.sh:115]: if this doesn't work, can you try adding `--user $(id -u):$(id -g)` to the `docker run` command

File: regression/upgrade/upgrade2.0.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+UpgradeTo=$2
+CurrentDirectory=$(cd `dirname $0` && pwd)
+FabricTestDir="$(echo $CurrentDirectory | awk -F'/fabric-test/' '{print $1}')/fabric-test"
+OperatorDir="$FabricTestDir"/tools/operator
+testdataDir="$FabricTestDir"/regression/testdata
+
+fabrictestCheckout(){
+    echo "Checking out to $1 branch"
+    git checkout -- .
+    rm go.sum
```
Comment (suryalnvs) [inline regression/upgrade/upgrade2.0.sh]: @btl5037 can you give your insights about this removing go.sum and then checking out from 1.4 to master

Comment (lindluni) [reply regression/upgrade/upgrade2.0.sh]: We need to either, fix the go.mod in the root of the repo or add a go.mod to the operator directory. I'm going to go for number 2, let me see if I can get that right

Comment (suryalnvs) [reply regression/upgrade/upgrade2.0.sh]: ok

Comment (lindluni) [reply regression/upgrade/upgrade2.0.sh]: If you rebase now, you shouldnt get regenerated go.sum's so you won't have to delete them

**Review by dongmingh — APPROVED** [review]

**Review by lindluni — APPROVED** [review]

**Pull request merged**